src/pulsar_producer.erl

```diff
@@ -129,8 +130,8 @@
 
 callback_mode() -> [state_functions, state_enter].
 
-start_link(PartitionTopic, Server, ProxyToBrokerUrl, ProducerOpts) ->
-    gen_statem:start_link(?MODULE, {PartitionTopic, Server, ProxyToBrokerUrl, ProducerOpts}, []).
+start_link(PartitionTopic, Server, ProxyToBrokerUrl, ParentPid, ProducerOpts) ->
+    gen_statem:start_link(?MODULE, {PartitionTopic, Server, ProxyToBrokerUrl, ParentPid, ProducerOpts}, []).
```

Comment:
Q: This is not public library API, right?

Reply:
Producers should be started by calling the APIs in `pulsar`, as the send APIs expect to use the handle returned by them.

Reply:
Reverted the arity change anyways after feedback.